### PR TITLE
new --cache switch for interface

### PR DESCRIPTION
**Pull request by davidkelly**
If you specify --cache <some directory> on the command line when launching, interface will use that directory for the ktx and assets cache.  So, one could pre-populate that cache and the initial experience upon launch _could_ be improved, say for events and so on.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1064/HighFidelity-Beta-PR10803-bda50af78098ea0fd549a201bcd0622906ecd23e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1064/HighFidelity-Beta-PR10803-bda50af78098ea0fd549a201bcd0622906ecd23e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1124/HighFidelity-Beta-PR10803-01155ba4e38e90800650a5a2324a61eeae61b15f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1124/HighFidelity-Beta-PR10803-01155ba4e38e90800650a5a2324a61eeae61b15f.exe

**Comment by davidkelly** [comment]
Test Plan

* install the PR.  
* it will launch (by default), that's fine.  Shut it down
* create a directory (initially empty) to use as the cache directory
* launch interface with `--cache <full path to that directory>`
* after launch, you will see the `data8`, `prepared`, and `ktx_cache` entries in the directory, and that they were written after launch
* in `AppData/Local/High Fidelity - PR10803/Interface`, you will see those same 3 directories, but the times will all be from the earlier launch

**Comment by CainFoool** [comment]
Testing.

**Comment by CainFoool** [comment]
# Test Passed
Test | Result
------- | --------
install the PR. | Pass
it will launch (by default), that's fine. Shut it down | Pass
create a directory (initially empty) to use as the cache directory | Pass
launch interface with --cache <full path to that directory> | Pass
after launch, you will see the data8, prepared, and ktx_cache entries in the directory, and that they were written after launch | Pass
in AppData/Local/High Fidelity - PR10803/Interface, you will see those same 3 directories, but the times will all be from the earlier launch | Pass

## New Cache Folder
image

## Default Cache Folder
image

Let me know if you'd like anything tested.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

File: interface/src/main.cpp [hunk 1]
```diff
@@ -96,6 +99,19 @@ int main(int argc, const char* argv[]) {
     if (allowMultipleInstances) {
         instanceMightBeRunning = false;
     }
+    if (parser.isSet(overrideAppLocalDataPathOption)) {
+        // get dir to use for cache
+        QString cacheDir = parser.value(overrideAppLocalDataPathOption);
+        if (!cacheDir.isEmpty()) {
```
Comment (zfox23) [inline interface/src/main.cpp]: Do we care about checks here to see if we can actually read to or write from the new directory?

File: interface/src/main.cpp [hunk 2]
```diff
@@ -96,6 +99,19 @@ int main(int argc, const char* argv[]) {
...
+            // tell everyone to use the right cache location
+            //
+            // this handles data8 and prepared
+            ResourceManager::setCacheDir(cacheDir);
+
+            // this does the ktx_cache
+            PathUtils::getAppLocalDataPath(cacheDir);
```
Comment (zfox23) [inline interface/src/main.cpp]: Wow. I'm glad it's this easy :D

File: interface/src/main.cpp (hunk 1 above)
Comment (davidkelly) [inline interface/src/main.cpp]: I was thinking it would maybe be good to do a check, but then never did :(

**Pull request merged**